pkg/git/detector/detector.go

```diff
@@ -0,0 +1,121 @@
+package detector
+
+import (
+	"fmt"
+	"github.com/redhat-developer/git-service/pkg/git/repository"
+	"regexp"
+	"sync"
+
+	"github.com/redhat-developer/git-service/pkg/git"
+	"github.com/redhat-developer/git-service/pkg/git/repository/github"
+)
+
+var gitServiceCreators = []repository.ServiceCreator{
+	github.NewRepoServiceIfMatches(),
+}
+
+func DetectBuildEnvironments(gitSource *git.Source) (*BuildEnvStats, error) {
```

Comment:
@MatousJobanek , you would be adding the 'output' of the detection to the `status` field of the `GitSource` CR, right?

Reply:
Yup, most likely there, yes. But not part of this PR :-)